Firestore/Source/Util/FSTDispatchQueue.h

```diff
@@ -56,12 +84,34 @@ NS_ASSUME_NONNULL_BEGIN
  * Schedules a callback after the specified delay.
  *
  * Unlike dispatchAsync: this method does not require you to dispatch to a different queue than
- * the current one (thus it is equivalent to a raw dispatch_after()).
+ * the current one.
+ *
+ * The returned FSTDelayedCallback handle can be used to cancel the callback prior to its running.
  *
  * @param block The block to run.
  * @param delay The delay (in seconds) after which to run the block.
+ * @param timerID An FSTTimerID that can be used from tests to check for the presence of this
+ *   callback or schedule it to run early.
+ * @return A FSTDelayedCallback instance that can be used for cancellation.
+ */
+- (FSTDelayedCallback *)dispatchAfterDelay:(NSTimeInterval)delay
+                                   timerID:(FSTTimerID)timerID
+                                     block:(void (^)(void))block;
+
+/**
+ * For Tests: Determine if a delayed callback with a particular FSTTimerID exists.
  */
-- (void)dispatchAfterDelay:(NSTimeInterval)delay block:(void (^)(void))block;
+- (BOOL)containsDelayedCallbackWithTimerID:(FSTTimerID)timerID;
+
+/**
+ * For Tests: Runs delayed callbacks early, blocking until completion.
+ *
+ * @param lastTimerID Only delayed callbacks up to and including this FSTTimerID will be run.
+ *   Method throws if no matching callback exists.
+ */
+- (void)runDelayedCallbacksUntil:(FSTTimerID)lastTimerID;
+
+- (void)removeDelayedCallback:(FSTDelayedCallback *)callback;
```

Comment:
Oh, hey. I just finally figured out what AppCode did when I used it to auto-fix a warning about this method the other day. :-)  

I moved it to the private interface in FSTDispatchQueue.mm though it's a tiny bit awkward because I have to declare it before FSTDelayedCallback, so now the FSTDispatchQueue private interface is split into two chunks.